package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -200,55 +251,54 @@ def _single_frame(self):
             self.results[dim][key] += hist
             self.results[dim][key_std] += np.square(hist)
 
-            key = 'char'
-            key_std = 'char_std'
+            key = 'charge_density'
+            key_std = 'charge_density_stddev'
             # histogram for positions weighted on charges
-            hist, _ = np.histogram(positions[:, idx],
-                                   weights=self.charges,
-                                   bins=self.nbins,
-                                   range=(0.0, max(self.dimensions)))
+            hist, bin_edges = np.histogram(positions[:, idx],
+                                           weights=self.charges,
+                                           bins=self.nbins,
+                                           range=(0.0, max(self.dimensions)))
 
             self.results[dim][key] += hist
             self.results[dim][key_std] += np.square(hist)
+            self.results[dim]['hist_bin_edges'] = bin_edges
 
     def _conclude(self):
-        k = 6.022e-1  # divide by avodagro and convert from A3 to cm3
+        avogadro = 6.022e23  # unit: mol^{-1}
```

Comment:
I considered this but have decided against it here because in units.py, it's defined as `6.02214129e+23` which would change the results. I changed it this way to make what is happening clearer while not affecting the results of the calculation. Happy to instead use the definition from units.py though if this is preferred.

Reply:
Not sure, @orbeckst you tend to have some stronger views on what should be in a minor versions vs a major one, can we call this a bug fix and make the switch for 2.2.0?

Reply:
I'd be in favor of using the consistent value from units.py. 

It does change results but it simply adds more significant digits so it doesn't warrant a major version in my opinion. I'd be ok calling it a fix.